attendance sheet correction/add_ids.py

```diff
@@ -0,0 +1,188 @@
+ 
+    percents = round(100.0 * count / float(total), 1)
+    bar = '=' * filled_len + ' ' * (bar_len - filled_len)
+ 
+    sys.stdout.write('%s: [%s] %s%s completed.\r' % (prefix, bar, percents, '%'))
+    sys.stdout.flush()
+
+# Levenschtein implementation ruthlessly borrowed from https://www.python-course.eu/levenshtein_distance.php
+def memoize(func):
+    mem = {}
+    def memoizer(*args, **kwargs):
+        key = str(args) + str(kwargs)
+        if key not in mem:
+            mem[key] = func(*args, **kwargs)
+        return mem[key]
+    return memoizer
+
+# This is fairly inefficient, but it works well for this purpose.
+@memoize    
+def levenshtein(s, t):
+    if s == "":
+        return len(t)
+    if t == "":
+        return len(s)
+    if s[-1] == t[-1]:
+        cost = 0
+    else:
+        cost = 1
+    
+    res = min([levenshtein(s[:-1], t)+1,
+               levenshtein(s, t[:-1])+1, 
+               levenshtein(s[:-1], t[:-1]) + cost])
+    return res
+
+def getSuggestions(name, students):
+    suggestions = []
+    for student in students.keys():
+        result = {"name":student, "editDistance":levenshtein(name, student)}
+        suggestions.append(result)
+
+    suggestions = sorted(suggestions, key=lambda item: item["editDistance"])
+    return suggestions[0:3]
+
+def importStudents():
+    students = {}
+    try:
+        with open('students.csv', 'r') as studentFile:
+            reader = csv.DictReader(studentFile)
+            for row in reader:
+                students["{0} {1}".format(row['First Name'], row['Last Name'])] = row['Student Key']
+
+    except Exception as e:
+        print("Error opening students.csv ({0}), exiting.".format(e))
+        sys.exit(1)
+
+    return students
+
+def importAttendance():
+    attendance = []
+    path = ""
+    if len(sys.argv) > 1: 
+        path = sys.argv[1]
+
+    else: # Prompt for file path
+        path = input("Please enter attendance sheet file name: ")
+
+    try:
+        with open(path, 'r') as attendanceFile:
+            reader = csv.reader(attendanceFile)
+            header = next(reader)
+            header.insert(3, "StudentKey")
+            attendance.append(header)
+            for row in reader:
+                row.insert(3, "")
+                attendance.append(row)
+
+    except Exception as e:
+        print("Error opening {0} ({1}), exiting.".format(path, e))
+        sys.exit(1)
+
+    return attendance, path
+
+def main():
+    # Set up our tables
+    students = importStudents()
+    attendance, fileName = importAttendance()
+    unmatched = {}
+
+    # Run through students and attempt to match them. Failing that, come up with suggestions.
+    total = len(attendance)
+    for i in range(1, len(attendance)):
+        progress(i, total, "Matching Student Names")
+        studentName = "{0} {1}".format(attendance[i][1], attendance[i][2])
+        if studentName in students.keys():
+            attendance[i][3] = students[studentName]
```

Comment:
I might just be missing something very simple, but in this line We've found a match! and should add the corresponding ID from students.csv to the attendance sheet, right? I'm confused why we are adding _students[studentName]_

Reply:
Nevermind-- just took a second look at importStudents func and I understand the _students_ data object now. _students[studentName]_ is indeed the students ID.